Enable bitrate limiting for outgoing video streams

jordanreedie opened the pull request:
Provides a function for limiting the bitrate of all outgoing video streams (barring screen streams)

There are two ways we can limit bitrate: SDP modification and `setParameters`

For SDP modification, the bitrate limit is set in the SDP exchange portion of connection negotiation. This means that if we want to change the bitrate limit we've set, we need to renegotiate the connection b/w two peers. 

Using `setParameters`, all we need to do is call the function with the appropriate parameters - there is no renegotiation required. This is, I believe, the preferred way of setting bitrate limits on the fly. However, it is only recently supported by browsers so requires the users to have up-to-date versions of chrome or firefox.

The first approach I took for bitrate limiting was SDP modification, which is why there are some unused settings & pieces of code that I've left in w/ comments. For now, since we want to adjust bitrate often, it is easier & more consistent to just use `setParameters`. It may be desirable in the future to support SDP modification for broader browser / device support (which is why I've left in some commented out code & unused settings), but as long as we advise users of our requirements it should be OK as a first step.

mlippert commented:
@jordanreedie There are a bunch of outdated packages in use, some w/ major version updates. Could you see about updating to the latest versions. I'd do it but I'm not familiar w/ building and testing this code.

```sh
$ npm outdated
Package            Current  Wanted  Latest  Location
attachmediastream    3.0.0     git     git  simplewebrtc
browserify          13.3.0  13.3.0  16.4.0  simplewebrtc
localmedia           6.0.0     git     git  simplewebrtc
socket.io-client     1.7.4   1.7.4   2.2.0  simplewebrtc
tape                4.10.1  4.11.0  4.11.0  simplewebrtc
uglify-js           2.8.29  2.8.29   3.6.0  simplewebrtc
webrtc-adapter       6.4.8   6.4.8   7.2.9  simplewebrtc
webrtc-testbed       0.0.1     git     git  simplewebrtc
wildemitter          1.2.0   1.2.1   1.2.1  simplewebrtc
```

Obviously we don't have to update the git dependencies at this time, although later we might want to check on updating packages in [rifflearning/attachMediaStream](https://github.com/rifflearning/attachMediaStream) and [rifflearning/localmedia](https://github.com/rifflearning/localmedia) repos too (and see if we want any updating in the upstream repo)

Commit:
add ability to limit the bitrate of video streams

Commit:
Update compiled simplewebrtc bundles

Commit:
Update jshint to not report on using functions before they're defined

mlippert committed:
trivial review changes

- Use single quotes for strings to match what is done in the rest of the file
- Use the instance logger instead of writing to the console directly
- Update the logged bitrate msg to include the bitrate being set

mlippert committed:
tell jslint that ecmascript 6 syntax is fine

jordanreedie commented:
@mlippert Yeah, I'll look into that. I know that at least one of them is going to be a significant undertaking / require pretty thorough testing (webrtc-adapter).

mlippert commented:
Oh right, yeah that major update should wait. Can you get the others though? With any luck the major changes won't actually affect this code.

mlippert commented:
I just force pushed having re-organized the changes into more targeted commits (keeping the updates to the compiled bundle files in their own commit). And then I added a commit w/ some trivial review changes (I've only reviewed 1 file so far).

mlippert reviewed (approved):
Everything looks good once the easy package updates have been done. I'm approving in anticipation.

jordanreedie committed:
update outdated packages, remove uglifyjs and request (unused)

jordanreedie committed:
rebuilt bundles w/ latest code

mlippert commented:
This was included in **Release** [1.2.0-dev.16](https://github.com/rifflearning/riff-rtc/releases/tag/1.2.0-dev.16) which was deployed to staging.riffplatform.com
